File: 0128-longest-consecutive-sequence/solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length < 2) {
            return nums.length;
        }

        Set<Integer> set = new HashSet<>();
        for (int i : nums) {
            set.add(i);
        }

        int max = Integer.MIN_VALUE;

        for (int i : set) {
            if (!set.contains(i - 1)) {
                int next = i + 1;
                while (set.contains(next))
                    next++;

                max = Math.max(next - i, max);
            }
        }

        return max;
    }
}

File: 0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        sety = set(nums)
        res = 0

        for num in sety:
            if num - 1 in sety:
                continue

            tmp = num
            while tmp in sety:
                tmp = tmp + 1

            res = max(res, tmp - num)

        return res

